full-node/sov-stf-runner/src/prover_service/blocking_prover.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use sov_rollup_interface::da::BlockHeaderTrait;
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::stf::StateTransitionFunction;
+use sov_rollup_interface::zk::ZkvmHost;
+
+use super::{Hash, ProverService, ProverServiceError};
+use crate::verifier::StateTransitionVerifier;
+use crate::{ProofGenConfig, Prover, RollupProverConfig, StateTransitionData};
+
+/// Prover that blocks the current thread and creates a ZKP proof.
+pub struct BlockingProver<StateRoot, Witness, Da, Vm, V>
```

Comment:
This part is a wrapper over existing proving logic and will be replaced by a multithreaded prover.